lib/WUI/http/httpd.c

```diff
@@ -2734,6 +2734,8 @@ static err_t http_find_file(struct http_state *hs, const char *uri, int is_09) {
         } else {
             uri = "401";
         }
+    } else {
+        uri = "404";
```

Comment:
Not sure how this works. I guess 404 is the response's status code, but why URI? And why a string?

Reply:
Peculiarities of the current web server implementation. Yes, we want to change the web server.

* The web server has some built-in „pages“. Some of them are specials and also invoke special status codes (like the 404 one). In fact, if you do `curl -v ip-of-the-printer/404`.
* By changing the URI and returning without providing the response, the web server loops back and uses that special page.

So, this is the very weird way the server works right now. Another reason to try to get rid of it ASAP.